runtime/experimental/makefiles/php.mk

```diff
@@ -0,0 +1,93 @@
+SHELL := /bin/bash
+.DEFAULT_GOAL := make_php
+
+url_php = https://github.com/php/php-src/archive/php-${VERSION_PHP}.tar.gz
+build_dir_php = ${DEPS}/php
+
+fetch_php:
+	mkdir -p ${build_dir_php}
+	${CURL} -Ls ${url_php} | tar $(shell ${TARGS} ${url_php}) ${build_dir_php} --strip-components=1
```

Comment:
I'm trying to wrap my head around everything. For example here why is this `${CURL}` and not simply `curl`?

What is the `${TARGS}` variable?

What is `$(shell ${TARGS} ${url_php})` resolving to?

Reply:
We set some baseline Environment Variables in the compiler container:

https://github.com/mnapoli/bref/blob/9ea33c127d7ff2055c5b1f18266e7ad7904ebb41/runtime/experimental/compiler.Dockerfile#L63-L82

`${TARGS}` contains the path to our [get_tar_args.sh helper script](https://github.com/mnapoli/bref/blob/9ea33c127d7ff2055c5b1f18266e7ad7904ebb41/runtime/experimental/helpers/get_tar_args.sh). That script determines what arguments `tar` will need in order to properly unpack the sourcecode. 
https://github.com/mnapoli/bref/blob/9ea33c127d7ff2055c5b1f18266e7ad7904ebb41/runtime/experimental/compiler.Dockerfile#L78

${CURL} contains the path to the `curl` exectable we compiled ourselves. 
https://github.com/mnapoli/bref/blob/9ea33c127d7ff2055c5b1f18266e7ad7904ebb41/runtime/experimental/compiler.Dockerfile#L79

Once we build our own Openssl, anything that relies on SSL communications can get funky, because we have two versions of Openssl on the system, and we have out environment setup to look for our own newly compiled libraries/executables first, then look for the older system libraries.

I ran into a handful of situations that got complicated because the old system curl got picked up, but couldn't find the the old openssl libraries (because of the way *LD_LIBRARY_PATH* is setup for development). So instead of relying on `$PATH` to magically find the correct binaries and libraries ... we explicitely use `/usr/bin/curl` for the system curl, prior to building our own curl, then we swap over to `${CURL}`

We use the `${CURL}` variable for our own path because it relies on the `${TARGET}` which could potentially be changed. If `${TARGET}` is modified, then `${CURL}` simply keeps working as advertised.

`$(shell ${TARGS} ${url_php})`
[$(shell ...) ](https://www.gnu.org/software/make/manual/html_node/Shell-Function.html)is a GNU Make Function.

It would resolve into `$(shell  /usr/local/bin/get_tar_args.sh https://github.com/php/php-src/archive/php-${VERSION_PHP}.tar.gz) which would return `xzC` and then make would run `${CURL} -Ls ${url_php} | tar xzC ${build_dir_php} --strip-components=1`